Fixed bug + extended features

Velover opened the pull request:
changed readme code to TS + new examples

added object.setStyle(style) to set the style with optional parameters
added object.createWithStyle(style, ...) to create a new object and set the style (wasn't able to do the factory pattern with the setStyle in types 
(```error: enable is not a valid member of void```)
)

added object.drawWithStyle(style, ...) to avoid creating additional lines during the quick tests

All new features were tested

-changed snake_case to camelCase
-removed style from the .d.ts
-changed fillOutStyle()
-added getStyle() because style was removed from the definition file

Velover committed:
added drawWithStyle
added setStyle
added OptionalStyle

Velover committed:
fixed the function fillOutStyle

Velover committed:
Turns gizmo into 1 liner during require  require(...Gizmo).enable()

Velover committed:
Changed the types for the one liner require + setStyle will return the object allowing one liner with setStyle

Velover committed:
setStyle doent return the type in the d.ts because it cannot track it's type maybe needed to convert the type into the class

added method createWithStyle

Velover committed:
Update README.md

updated code for TypeScript and added new functions in the description

Velover committed:
Update README.md

christopher-buss commented on src/gizmo.d.ts:
```diff
@@ -23,8 +28,8 @@ declare interface GizmoClass<T extends (...args: Array<any>) => void> {
 declare namespace Gizmo {
 	export const style: Style;
 
-	export function enable(): void;
```
I'm unsure why we want to chain here. Typically you'd only call enable() once which allows all gizmos in the game to be drawn, or disable() to stop them, and it should probably be done in one central place. Gizmo's also shouldn't be left enabled during a live game, this is debug only code, so I think enabling Gizmo on require is an anti-pattern anyway.

christopher-buss commented on src/gizmo.luau:
```diff
@@ -356,6 +365,14 @@ end
 -- Gizmo Class Wrapper
 -------------------------------------------------------------------------------
 
+local function fillOutStyle(optional_style: OptionalStyle)
```
Can we remove the snake_case naming in favour for camelCase as the rest of the file is already setup this way.

Velover replied:
sure
I haven't noticed that I wrote like that 😅

christopher-buss commented on src/gizmo.d.ts:
```diff
@@ -9,11 +9,16 @@ declare type Style = {
 	disable(): void;
 	draw: (...args: Parameters<T>) => void;
+	drawWithStyle: (optional_style: OptionalStyle, ...args: Parameters<T>) => void;
+	setStyle(optional_style: OptionalStyle): void;
```
camelCase please!

christopher-buss commented on README.md:
```diff
@@ -26,22 +26,40 @@ Renders the gizmo for a single frame
 
 Updates the gizmos appearance
 
 #### object.style
```
I'm happy with setStyle, it makes the ergonomics of setting easier as we can just default to the main globalStyle object. Could we actually remove direct access to the style object? We'd need a setGlobalStyle function on the exports, but I don't have an issue with this. We'd also likely want to add a corresponding getStyle function for completion sake.

I think it makes more sense to go this road than having the ability to set the style manually (which doesn't fill), and the setStyle function because it's two different ways to do similar things that aren't quite the same.

Velover replied:
yeah i will add that

Velover committed:
Removed direct assigning style in favor of setStyle() and snake_case was swapped to camelCase

Velover committed:
snake_case was swapped to camelCase
removed returning of export_table in enable / disable
added additional check in the fillOutStyle to avoid cloning globalStyle if the gizmo is not active

Velover committed:
added the function to get Style because it was removed from the definition file

Velover committed:
Added getStyle to definition file

Velover committed:
Merge branch 'main' of https://github.com/Velover/gizmo_plus
